doc/template/HOW-TO.md

```diff
@@ -0,0 +1,128 @@
+    $ git remote -v
+    origin  https://github.com/swcarpentry/lesson-example.git (fetch)
+    origin  hhttps://github.com/swcarpentry/lesson-example.git (push)
+    template        https://github.com/r-gaia-cs/swc-lesson-template.git (fetch)
+    template        https://github.com/r-gaia-cs/swc-lesson-template.git (push)
+    template-upstream       https://github.com/swcarpentry/lesson-template.git (fetch)
+    template-upstream       https://github.com/swcarpentry/lesson-template.git (push)
+
+I use our example because it is easy to debug but if you want to use
+`origin` as your fork of the lessons you are working is totally OK.
+
+## Preparation
+
+Every time before I start to hack the template I update my local copy using
+
+    $ git fetch origin
+    $ git fetch template
+    $ git fetch template-upstream
+
+Once my local copy is updated I create a feature branch,
+**starting from the last commit on `lesson-example`**, where I will work.
+
+  $ git checkout -b example-centering-image origin/gh-pages
+
+## Hacking
+
+For this how-to, suppose that I want to add the class `img-responsive` to every `img` elements of our lessons.
+One **quickly and dirty** way to do that is change `Makefile` with
+
+    $ git diff
+    diff --git a/Makefile b/Makefile
+    index 4d38850..ff49ca9 100644
+    --- a/Makefile
+    +++ b/Makefile
+    @@ -55,6 +55,7 @@ motivation.html : motivation.md _layouts/slides.revealjs Makefile
+            --filter=tools/filters/id4glossary.py \
+            $(INCLUDES) \
+            -o $@ $<
+    +       sed -i 's/^<img /<img class="img-responsive" /g' $@
+
+     # Pattern to convert R Markdown to Markdown.
+     %.md: %.Rmd $(R_CHUNK_OPTS)
+
+To avoid losing my work, I save it in the staging area:
+
+    $ git add -u
+
+Now I compile the HTML files:
+
+    $ make -B preview
+
+And verify the changes.
+
+    $ git diff
+    diff --git a/01-one.html b/01-one.html
+    index 7622e55..1fc0d45 100644
+    --- a/01-one.html
+    +++ b/01-one.html
+    @@ -56,7 +56,8 @@ program</code></pre>
+     </div>
+     <p>or an image from the <code>figs</code> directory:</p>
+     <div class="figure">
+    -<img src="fig/example.svg" title="this is the image&#39;s alt text" alt="this is the image&#39;s title" /><p class="caption">this is the image's title</p>
+    +<img class="img-responsive" src="fig/example.svg" title="this is the image&#39;s alt text" alt="this is the image&#39;s title" />
+    +<p class="caption">this is the image's title</p>
+     </div>
+     <div id="challenge-title" class="challenge panel panel-success">
+     <div class="panel-heading">
+
+This is exactly what I want. Great!!! I can now commit my changes.
+
+    $ git commit -m 'Add img-responsive class to img'
+    [example-centering-image a9cc86d] Add img-responsive class to img
+     1 file changed, 1 insertion(+)
+
+I will also save the change to the HTML file.
+
+    $ git add -u
```

Comment:
With my explicit-file approach, I'd use `git add *.html` here.  Alternatively, we could skip the separate `add` and use `git commit -am 'Update HTML'`, since that's discussed in the git-novice lesson.